compliance-api/src/compliance_api/services/case_file.py

```diff
@@ -13,3 +15,89 @@ def get_initiation_options(cls):
             {"id": case.name, "name": CASE_FILE_INITIATION_MAP[case]}
             for case in CaseFileInitiationEnum
         ]
+
+    @classmethod
+    def get_all_case_files(cls):
+        """Return all the case files."""
+        return CaseFile.get_all()
+
+    @classmethod
+    def get_case_file_by_id(cls, case_file_id: int):
+        """Return case file by id."""
+        return CaseFile.find_by_id(case_file_id)
+
+    @classmethod
+    def create_case_file(cls, case_file_data: dict):
+        """Create case file."""
+        case_file_obj = _create_case_file_object(case_file_data)
+        existing_case_file = cls.get_case_file_by_file_number(
+            case_file_obj.get("case_file_number", None)
+        )
+        if existing_case_file:
+            raise ResourceExistsError(
+                f"Case file with the number {case_file_obj['case_file_number']} exists"
+            )
+        with session_scope() as session:
+            created_case_file = CaseFile.create_case_file(case_file_obj, session)
+            cls.insert_or_update_officers(
+                created_case_file.id, case_file_data.get("officer_ids", None), session
+            )
+        return created_case_file
+
+    @classmethod
+    def update_case_file(cls, case_file_id: int, case_file_data: dict):
+        """Update case file."""
+        case_file_obj = _create_case_file_object(case_file_data)
+        existing_case_file = cls.get_case_file_by_file_number(
+            case_file_obj.get("case_file_number", None)
+        )
+        if existing_case_file and existing_case_file.id != case_file_id:
+            raise ResourceExistsError(
+                f"Case file with the number {case_file_obj['case_file_number']} exists"
+            )
+        with session_scope() as session:
+            updated_case_file = CaseFile.update_case_file(
```

Comment:
CaseFile can be called as CaseFileModel or CaseFileService ; will be easier to read